pkg/lib/image/image.go

```diff
@@ -13,5 +13,12 @@ func InferImagePullPolicy(image string) corev1.PullPolicy {
 	if strings.Contains(image, "@") {
 		return corev1.PullIfNotPresent
 	}
+
+	// Ensure test registry images are pulled only if needed
+	// These will normally be loaded through kind load docker-image
+	// This is also helps support the e2e fixture image re-build flow (see e2e GHA)
+	if strings.HasPrefix(image, "quay.io/olmtest/") {
+		return corev1.PullIfNotPresent
+	}
```

Comment:
Same - the only other way I could think of is by threading another command line arg down the stack - idwtdt 

I figured this isn't so bad, since the pattern should only ever match _our_ registry. So, there shouldn't be any hidden surprises.

But, yeah, I don't like it either. But anything else that I could think of felt like too many changes T_T

:thinking: let me see if I can programmatically grab the sha from the image:tag - I think we already have a dependency on the docker client anyway